src/containers/TopicArticlePage/components/TopicArticleConnections.jsx

```diff
@@ -0,0 +1,184 @@
+      }
+      return {
+        openedPaths,
+      };
+    });
+  }
+
+  async addToTopic(id, closeModal) {
+    this.props.stageTaxonomyChanges({ addTopicId: id });
+    closeModal();
+  }
+
+  async addTopic(id, path, closeModal) {
+    this.props.stageTaxonomyChanges({ path });
+    closeModal();
+  }
+
+  toggleFilter(id) {
+    this.setState(({ activeFilters }) => ({
+      activeFilters: activeFilters.includes(id)
+        ? activeFilters.filter(activeFilter => activeFilter !== id)
+        : [...activeFilters, id],
+    }));
+  }
+
+  render() {
+    const {
+      t,
+      structure,
+      availableFilters,
+      activeTopics,
+      ...rest
+    } = this.props;
+    const { activeFilters, openedPaths } = this.state;
+
+    return (
+      <Fragment>
+        <FieldHeader
+          title={t('taxonomy.topics.title')}
+          subTitle={t('taxonomy.topics.subTitleTopic')}>
+          <HowToHelper
+            pageId="TaxonomyTopicConnections"
+            tooltip={t('taxonomy.topics.helpLabel')}
+          />
+        </FieldHeader>
+        <ActiveTopicConnections
+          activeTopics={activeTopics}
+          type="topic-article"
+          {...rest}
+        />
+        <Modal
+          backgroundColor="white"
+          animation="subtle"
+          size="large"
+          narrow
+          minHeight="85vh"
+          activateButton={
+            <Button>{t('taxonomy.topics.filestructureButton')}</Button>
+          }>
+          {closeModal => (
+            <Fragment>
+              <ModalHeader>
+                <ModalCloseButton
+                  title={t('taxonomy.topics.filestructureClose')}
+                  onClick={closeModal}
+                />
+              </ModalHeader>
+              <ModalBody>
+                <StyledTitleModal>
+                  {t('taxonomy.topics.filestructureHeading')}:
+                </StyledTitleModal>
+                <hr />
+                <Structure
+                  openedPaths={openedPaths}
+                  structure={structure}
+                  toggleOpen={this.handleOpenToggle}
+                  renderListItems={props => (
+                    <StructureFunctionButtons
+                      {...props}
+                      activeTopics={activeTopics}
+                      availableFilters={availableFilters}
+                      activeFilters={activeFilters}
+                      toggleFilter={this.toggleFilter}
+                      addToTopic={() => this.addToTopic(props.id, closeModal)}
+                      addTopic={() =>
+                        this.addTopic(props.id, props.path, closeModal)
+                      }
+                    />
+                  )}
+                  activeFilters={activeFilters}
+                  filters={availableFilters}
+                />
+              </ModalBody>
+            </Fragment>
+          )}
+        </Modal>
+      </Fragment>
+    );
+  }
+}
+
+TopicArticleConnections.propTypes = {
```

Comment:
Nesten alt optional med vilje?

Reply:
Ikke at det kræsjer uansett om det er required eller ikke, men nå får man jo ikke warning en gang? Noen ting er jo faktisk required for at det skal fungere.